src/Compilers/CSharp/Test/Syntax/Parsing/MemberDeclarationParsingTests.cs

```diff
@@ -10472,6 +10472,1942 @@ public class Class
             EOF();
         }
 
+        [Theory, WorkItem("https://github.com/dotnet/roslyn/issues/74482")]
+        [InlineData("class", SyntaxKind.ClassDeclaration, SyntaxKind.ClassKeyword)]
+        [InlineData("struct", SyntaxKind.StructDeclaration, SyntaxKind.StructKeyword)]
+        [InlineData("interface", SyntaxKind.InterfaceDeclaration, SyntaxKind.InterfaceKeyword)]
+        [InlineData("record", SyntaxKind.RecordDeclaration, SyntaxKind.RecordKeyword)]
+        [InlineData("record struct", SyntaxKind.RecordStructDeclaration, SyntaxKind.RecordKeyword)]
+        [InlineData("record class", SyntaxKind.RecordDeclaration, SyntaxKind.RecordKeyword)]
+        public void ExtraCloseCurly_TypeWithSingleInvalidMemberFollowing(string typeKind, SyntaxKind typeSyntaxKind, SyntaxKind keywordKind)
+        {
+            var text = $$"""
+                namespace N
+                {
+                    {{typeKind}} Type
+                    {
+                    }
+
+                    // One method that will move into the type above.
+                    private void M()
+                    {
+                    }
+                }
+                """;
+            UsingTree(text,
+                // (5,5): error CS1519: Invalid token '}' in class, record, struct, or interface member declaration
```

Comment:
this is fair.  i'll think on it too and check in with you later.

Reply:
having thought about it, i'm ok with this error message.  it is invalid.  Now it happens to be invalid because of what we see later, btu i'm ok with that.

note: i'm not pushing back strongly.  if you do want a specific error here, lmk.

Reply:
> A namespace cannot directly contain members such as fields, methods or statements.

i'm not a fan of this *primarily* because the tree we produce won't jive with this.  The node will appear inside the class from all other perspectives.  But i do think we can clarify the error we put on the `}`.  Something like `Errant '}' found between type members`.